algorithms/src/std_algorithms/Kokkos_CopyIf.hpp

```diff
@@ -54,7 +54,8 @@ template <
     std::enable_if_t<::Kokkos::is_execution_space_v<ExecutionSpace>, int> = 0>
 auto copy_if(const ExecutionSpace& ex,
              const ::Kokkos::View<DataType1, Properties1...>& source,
-             ::Kokkos::View<DataType2, Properties2...>& dest, Predicate pred) {
+             const ::Kokkos::View<DataType2, Properties2...>& dest,
+             Predicate pred) {
```

Comment:
what are the preconditions?  we don't allocate and expect that the destination view has enough storage?

Reply:
we don't resize the destination view to fit